### PR TITLE
GH-189 Adds Rest methods to query baseUrl

### LINKED ISSUE #189: Provide support for querying configured baseUrl
As a developer
I want to query Rest's `baseUrl` without explicitly providing any path
So that it it possible to do so

Proposal is to add no-argument get, post, put, etc methods which will call preconfigured baseUrl or fail if there is no one configured.

### PR DESCRIPTION
**Pull request by AlexanderYastrebov**
Fixes #189

### CODE REVIEW

File: riptide-core/src/main/java/org/zalando/riptide/RequestArguments.java [hunk 1]
```diff
@@ -52,15 +52,34 @@
     RequestArguments withBody(@Nullable Object body);
 
     default RequestArguments withRequestUri() {
-        // expand uri template
-        final URI uri = Optional.ofNullable(getUri())
-                .orElseGet(() -> fromUriString(getUriTemplate())
+        final URI uri = getUri();
+
+        final URI toResolve;
+        if (uri != null) {
```
Comment (whiskeysierra) [inline riptide-core/src/main/java/org/zalando/riptide/RequestArguments.java]: I'd invert both ifs.

File: riptide-core/src/main/java/org/zalando/riptide/RequestArguments.java [hunk 2]
```diff
@@ -52,15 +52,34 @@
...
+            }
+        }
 
-        // resolve uri against base url
-        final URI resolved = getBaseUrl() == null ? uri : getBaseUrl().resolve(uri);
+        final URI baseUrl = getBaseUrl();
+
+        final URI resolved;
+        if (toResolve == null) {
+            resolved = checkNotNull(baseUrl, "base url required");
```
Comment (whiskeysierra) [inline riptide-core/src/main/java/org/zalando/riptide/RequestArguments.java]: Illegal argument or illegal state would also fit.

**Comment by AlexanderYastrebov** [comment]
It is not the first time I see failed build due ti `TemporaryExceptionPluginTest.shouldClassifyAsPermanent`
Last time I just closed/reopened PR to retrigger build and it had gone

File: riptide-core/src/main/java/org/zalando/riptide/RequestArguments.java (hunk 1 above)
Comment (AlexanderYastrebov) [inline riptide-core/src/main/java/org/zalando/riptide/RequestArguments.java]: :)

File: riptide-core/src/main/java/org/zalando/riptide/RequestArguments.java (hunk 2 above)
Comment (AlexanderYastrebov) [inline riptide-core/src/main/java/org/zalando/riptide/RequestArguments.java]: I decided to use the same Preconditions.checkNotNull as everywhere else.

**Comment by AlexanderYastrebov** [comment]
Raised issue regarding of build failure #197

**Comment by whiskeysierra** [comment]
Much better now :stuck_out_tongue_winking_eye:

**Comment by AlexanderYastrebov** [comment]
:+1:

**Pull request merged**